lib/ex_sdp/media.ex

```diff
@@ -65,15 +67,14 @@ defmodule ExSDP.Media do
   end
 
   @spec add_attribute(media :: t(), attribute :: Attribute.t()) :: t()
-  def add_attribute(media, attribute) do
-    attributes = media.attributes ++ [attribute]
-    Bunch.Struct.put_in(media, :attributes, attributes)
-  end
+  def add_attribute(media, attribute), do: Map.update!(media, :attributes, &(&1 ++ [attribute]))
 
-  @spec get_attribute(media :: t(), key :: String.t()) :: Attribute.t()
+  @spec get_attribute(media :: t(), key :: ExSDP.Attribute.RTPMapping.t() | Attribute.key()) ::
+          Attribute.t()
   def get_attribute(media, key) do
     media.attributes
     |> Enum.find(fn
+      %ExSDP.Attribute.RTPMapping{} -> true
```

Comment:
This doesn't look too good but I will fix it in next PR after introducing more struct attributes.

Reply:
doesn't this find RTPMapping regardless the key? maybe
```suggestion
      %module{} -> key == module
```

Reply:
This looks great, didn't know we can do `%module{}`. Regarding searching by `:rtpmap`. There was such possibility but I have removed it. In previous version to add RTPMapping attribute we had to do this
```elixir
 attributes: [
          %Attribute{
            key: :rtpmap,
            value: %Attribute.RTPMapping{
              clock_rate: 90_000,
              encoding: "h263-1998",
              params: nil,
              payload_type: 99
            }
          }
        ]
```

now we can do
```elixir
 attributes: [
          %Attribute.RTPMapping{
            clock_rate: 90_000,
            encoding: "h263-1998",
            params: nil,
            payload_type: 99
          }
        ]
```  

IMO searching by module name i.e. `Media.get_attribute(media, RTPMapping)` is pretty cool.

Reply:
The point is that when some attributes are searched by atom, and others by struct name, there's no uniform way to find an attribute. I'm not saying that we should return to the previous way, we could just store the mapping attribute name -> struct somewhere